src/IndexerServiceProvider.php

```diff
@@ -2,11 +2,28 @@
 
 namespace AwStudio\Indexer;
 
-use AwStudio\Indexer\Commands\CreateIndexCommand;
 use Illuminate\Support\ServiceProvider;
+use AwStudio\Indexer\Commands\RunCommand;
+use AwStudio\Indexer\Commands\CreateIndexCommand;
+use AwStudio\Indexer\Contracts\HtmlLoader;
 
 class IndexerServiceProvider extends ServiceProvider
 {
+    /**
+     * Register application services.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->registerHtmlLoader();
+        $this->registerRunCommand();
+
+        if ($this->app->runningInConsole()) {
+            $this->commands(['indexer.commands.run']);
+        }
+    }
```

Comment:
Service and command registrations belong into the register method not the boot method of the service provider.